drivers/ever-hid.c

```diff
@@ -0,0 +1,638 @@
+		n = snprintfcat(line, sizeof(line), n ? ":%02x" : "%02x",
+			((unsigned char *)buf)[i]);
+	}
+
+	return line;
+}
+
+static const char *ever_ip_address_fun(double value)
+{
+	static int report_counter = 1;
+	int report_id = 211;
+
+	if(report_counter == 1) 
+		report_id = 211; /* notification dest ip */
+	else if(report_counter == 2)
+		report_id = 230; /* ip address */
+	else if(report_counter == 3)
+		report_id = 231;	/* network mask */
+	else if(report_counter == 4)
+		report_id = 232;	/* default gateway */
+
+	report_counter== 4 ? report_counter=1 : report_counter++; 
+
+	int len = reportbuf->len[report_id];
+	const void *buf = reportbuf->data[report_id];
+
+	static char line[100];
+	line[0] = '\0';
+	int n = 0;	/* number of characters currently in line */
+	int i;	/* number of bytes output from buffer */
+
+	/*skip first element which is a report id */
+	for (i = 1; i < len; i++) 
+	{
+		n = snprintfcat(line, sizeof(line), n ? ".%d" : "%d", ((unsigned char *)buf)[i]);
+	}
+
+	return line;
+}
+
+static const char *ever_packets_fun(double value)
+{
+	static int report_counter = 1;
+	int report_id = 215;
+
+	if(report_counter == 1 )
+		report_id = 215;
+	else if(report_counter == 2 )
+		report_id = 216;
+	else if(report_counter == 3 )
+		report_id = 217;
+	else if(report_counter == 4 )
+		report_id = 218;
+
+	report_counter== 4 ? report_counter=1 : report_counter++; 
+
+	int len = reportbuf->len[report_id];
+	const unsigned char *buf = reportbuf->data[report_id];
+
+	static char line[100];
+	line[0] = '\0';
+
+	/*skip first elemnt which is a report id */
+
+	if(len < 5)
+		return "";
+
+	int res = (int)buf[1];
+	res |= (int)buf[2] << 8;
+	res |= (int)buf[3] << 16;
+	res |= (int)buf[4] << 24;
+
+    snprintf(line, sizeof(line), "%d", res);
+	return line;
+
+}
+
+static const char* ever_workmode_fun(double value)
+{
+	int workmode_report_id = 74;
+	const unsigned char *buf = reportbuf->data[workmode_report_id];
+
+	static char line[100];
+	line[0] = '\0';
+
+	/*skip first element which is a report id */
+	snprintfcat(line, sizeof(line), "%d", buf[1]);
+	
+
+
+	int workmode = atoi(line);
+
+	switch(workmode)
+	{
+		case 1:
+			return "UNKNOWN";
+		case 2:
+			return "STOP";
+		case 4:
+			return "ONLINE";
+		case 8:
+			return "ONBATTERY";
+		case 16:
+			return "WATCH";
+		case 32:
+			return "WAITING";
+		case 64:
+			return "EMERGENCY";
+		default:
+			return "UNKNOWN";
+
+	}
+	
+}
+
+static const char* ever_messages_fun(double value)
+{
+	int messages_report_id = 75;
+	const unsigned char *buf = reportbuf->data[messages_report_id];
+
+	static char line[200];
+	line[0] = '\0';
+
+	/*skip first element which is a report id */
+	int messages = (int)buf[1];
+	messages |= (int)buf[2] << 8;
+
+	int n = 0;	/* number of characters currently in line */
+
+	if(messages & 0x01)
+		n = snprintfcat(line, sizeof(line), n ? " %s" : "%s", "OVERLOAD");
+	if(messages & 0x02)
+		n = snprintfcat(line, sizeof(line), n ? " %s" : "%s", "BATTERY_LOW");
```

Comment:
seems to be a duplicate of `ups.status: OB LB`

Reply:
Updated later: these two matches are commented away